pkg/kv/kvserver/store.go

```diff
@@ -1608,6 +1610,7 @@ func NewStore(
 			s.scanner.AddQueues(s.tsMaintenanceQueue)
 		}
 	}
+	s.rangefeedRestarter = newRangefeedRestarter(s)
```

Comment:
Is there a particular reason why we construct this so much earlier than the start, unlike the scheduler which is constructed and started simultaneously?